internal/fs/fs_reader_test.go

```diff
@@ -151,8 +152,8 @@ func verifyDirectoryContentsFI(t testing.TB, fs FS, dir string, want []os.FileIn
 }
 
 func checkFileInfo(t testing.TB, fi os.FileInfo, filename string, modtime time.Time, mode os.FileMode, isdir bool) {
-	if fi.IsDir() {
-		t.Errorf("IsDir returned true, want false")
+	if fi.IsDir() != isdir {
+		t.Errorf("IsDir returned %t, want %t", fi.IsDir(), isdir)
```

Comment:
We need another test, `os.FileInfo` defines that `Name()` returns the base of the file name, so `path.Base(filename)`:
```suggestion
		t.Errorf("IsDir returned %t, want %t", fi.IsDir(), isdir)
	}
	
	if path.Base(fi.Name()) != fi.Name() {
		t.Errorf("Name() returned is not base: want %q, got %q", path.Base(fi.Name()), fi.Name())
```

Reply:
We also need to check `fi.Name() != path.Base(filename)` below (unfortunately I can't add a suggestion for it).

Reply:
Ah yes, I completely missed that, thanks!